### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by adrien-berchet**
Here is a first version of the refactoring I mentioned to be able to store several files and fetch their dataset hierarchies.
It's currently not working, I have an issue with borrowing mutable and immutable reference but I will see that later (or @anilbey  you know how to fix it? :innocent: ).
Does this refactoring makes sense to you?

### CODE REVIEW

**Comment by adrien-berchet** [comment]
Note that I let a few useless stuff that I will have to clean later.

**Review by anilbey — COMMENTED** [review]
Fantastic change!
I left just a small comment.

To be honest, I couldn't fully understand the borrowing issue.

File: src/gui.rs [hunk 1]
```diff
@@ -123,7 +137,13 @@ impl eframe::App for NWBView {
 
         // Collect dropped files:
         if !ctx.input().raw.dropped_files.is_empty() {
-            self.dropped_files = ctx.input().raw.dropped_files.clone();
+            for file in ctx.input().raw.dropped_files.clone() {
+                match file.path {
+                    None => println!("Could not load the file!"),
+                    Some(x) => (self.loaded_files)
+                        .push(hdf::read_nwb_file(&x.display().to_string()).unwrap()),
```
[the remark this answers is not in the document]
Comment (adrien-berchet) [reply src/gui.rs:144]: Ah yes yes, I forgot a match here. I will fix that.

[the remark this answers is not in the document]
Comment (adrien-berchet) [reply src/gui.rs]: Should be good now.

**Pull request merged**